### DISCUSSION TITLE
Return Join data on.subscribe()

### DISCUSSION THREAD

**Discussion by brentrobbins** [supabase/supabase#6455] [Questions] [answered]
Is it possible to include join data in the new payload when on.subscribe() is triggered?

**Comment by GaryAustin1** [comment, marked as answer]
No, you will need to do a select request to get the rest of your data in the payload handler, or add critical data to your table being monitored by realtime.

**Comment by brentrobbins** [comment]
OK, thanks @GaryAustin1 for the help.

As for my situation, I have a messages feature and I have the on.subscribe() listening to newly inserted messages. This works great, however I have a `user_id` field that's joined with a `users` table and I would like to get that user data with the payload or sort out the best way to update the state to show the new message along with the author/user's info.

**Reply by GaryAustin1** [reply]
The simplest way is when you get the payload, do a select back to the database and merge that data with the payload object you have.

**Comment by LucasAbijmil** [comment]
Sorry to bring this old topic to light, but what's the situation today? 
On iOS, the SDK uses v2 of realtime - will we be able to make JOIN via this update or should we use the proposed solution which would be to query DB again?

Thanks

**Reply by GaryAustin1** [reply]
There is no join currently in realtime.

**Comment by kevin4dhd** [comment]
is it possible to use a view to solve this problem?

**Reply by GaryAustin1** [reply]
Realtime does not work on views.

**Reply by kevin4dhd** [reply]
I'm thinking of connecting it to hasura, do you think it's a good idea to solve this problem?

**Reply by GaryAustin1** [reply]
I don't know anything about Hasura or how it would help with realtime.
